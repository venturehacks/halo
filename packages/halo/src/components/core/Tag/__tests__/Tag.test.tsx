import { mount, render } from 'enzyme';
import React from 'react';

import { Tag } from '../index';

describe('Tag', () => {
  test('smoke', () => {
    const component = render(<Tag label="PHP" />);
    expect(component.text()).toEqual('PHP');
  });

  test('tag callback', () => {
    const callback = jest.fn();
    const wrapper = mount(<Tag label="click me" onClick={callback} />);

    expect(callback).toHaveBeenCalledTimes(0);
    wrapper.simulate('click');
    expect(callback).toHaveBeenCalledTimes(1);
  });

  test('close button callback', () => {
    const tagCallback = jest.fn();
    const closeCallback = jest.fn();
    const wrapper = mount(
      <Tag
        label="I can be closed"
        onClick={tagCallback}
        onClose={closeCallback}
      />,
    );

    const closeButton = wrapper.find('svg');

    // clicking on close button should not run parent onClick
    closeButton.simulate('click');
    expect(closeCallback).toHaveBeenCalledTimes(1);
    expect(tagCallback).toHaveBeenCalledTimes(0);

    // parent onClick should not trigger onClose
    wrapper.simulate('click');
    expect(closeCallback).toHaveBeenCalledTimes(1);
    expect(tagCallback).toHaveBeenCalledTimes(1);
  });

  describe('snapshot', () => {
    test('plain', () => {
      const component = render(<Tag label="plain" />);
      expect(component).toMatchSnapshot();
    });

    test('with children', () => {
      const component = render(<Tag>with children</Tag>);
      expect(component).toMatchSnapshot();
      expect(component.text()).toEqual('with children');
    });

    describe('color', () => {
      test('red', () => {
        const component = render(<Tag color="red" />);
        expect(component).toMatchSnapshot();
      });

      test('gray', () => {
        const component = render(<Tag color="gray" />);
        expect(component).toMatchSnapshot();
      });
    });

    describe('size', () => {
      test('xs', () => {
        const component = render(<Tag size="xs" />);
        expect(component).toMatchSnapshot();
      });

      test('sm [default]', () => {
        const component = render(<Tag size="sm" />);
        expect(component).toMatchSnapshot();
      });

      test('md', () => {
        const component = render(<Tag size="md" />);
        expect(component).toMatchSnapshot();
      });
    });
  });
});
