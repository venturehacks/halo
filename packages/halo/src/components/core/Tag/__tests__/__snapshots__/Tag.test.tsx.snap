// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag snapshot color gray 1`] = `
<span
  class="inline-flex flex-row items-center mr-2 last:mr-0 bg-slate-100 text-dark-aaa rounded-md gap-2 text-xs px-3 py-1 bg-opacity-75"
/>
`;

exports[`Tag snapshot color red 1`] = `
<span
  class="inline-flex flex-row items-center mr-2 last:mr-0 bg-red-100 text-red-600 rounded-md gap-2 text-xs px-3 py-1 bg-opacity-75"
/>
`;

exports[`Tag snapshot plain 1`] = `
<span
  class="inline-flex flex-row items-center mr-2 last:mr-0 bg-blue-100 text-dark-aaaa rounded-md gap-2 text-xs px-3 py-1 bg-opacity-75"
>
  plain
</span>
`;

exports[`Tag snapshot shape pill 1`] = `
<span
  class="inline-flex flex-row items-center mr-2 last:mr-0 bg-blue-100 text-dark-aaaa rounded-full gap-2 text-xs px-3 py-1 bg-opacity-75"
/>
`;

exports[`Tag snapshot shape rectangle 1`] = `
<span
  class="inline-flex flex-row items-center mr-2 last:mr-0 bg-blue-100 text-dark-aaaa rounded-md gap-2 text-xs px-3 py-1 bg-opacity-75"
/>
`;

exports[`Tag snapshot size 2xs 1`] = `
<span
  class="inline-flex flex-row items-center mr-2 last:mr-0 bg-blue-100 text-dark-aaaa rounded-md gap-1 text-3xs px-2 py-0.5 antialiased bg-opacity-75"
/>
`;

exports[`Tag snapshot size md 1`] = `
<span
  class="inline-flex flex-row items-center mr-2 last:mr-0 bg-blue-100 text-dark-aaaa rounded-md gap-2 text-md px-3 py-2 bg-opacity-75"
/>
`;

exports[`Tag snapshot size sm [default] 1`] = `
<span
  class="inline-flex flex-row items-center mr-2 last:mr-0 bg-blue-100 text-dark-aaaa rounded-md gap-2 text-xs px-3 py-1 bg-opacity-75"
/>
`;

exports[`Tag snapshot size xs 1`] = `
<span
  class="inline-flex flex-row items-center mr-2 last:mr-0 bg-blue-100 text-dark-aaaa rounded-md gap-1 text-2xs px-2 py-0.5 antialiased bg-opacity-75"
/>
`;

exports[`Tag snapshot with children 1`] = `
<span
  class="inline-flex flex-row items-center mr-2 last:mr-0 bg-blue-100 text-dark-aaaa rounded-md gap-2 text-xs px-3 py-1 bg-opacity-75"
>
  with children
</span>
`;
